book/10-b-cabbage.md

```diff
@@ -3,372 +3,522 @@
+
+>Closing a file will not stop it from performing. To stop a file from performing you must hit the Stop button.  
+
+<a name="CreatingGUIs"></a>
+
+## Creating GUI interfaces for instruments
+
+
+To create a GUI for your instrument you must enter edit mode for that instrument. You can do this by hitting the Edit mode button at the top of the file's editor, or by hitting Ctrl+e when the editor for that instrument have focus. Once in edit mode, each widget will have a thin white border around it. you can move widgets around whilst in edit. You can also right-click and insert new widgets, as well as modify their appearance using the GUI properties editor on the right-hand side of the screen. 
+
+![](../resources/images/10-b-cabbageeditmode.gif)
+
+You will notice that when you select a widget whilst in edit mode, Cabbage will highlight the corresponding line of text in your source code. When updating GUI properties, hit 'Enter' when you are editing single line text or numeric properties, 'Tab' when you are editing multi-line text properties, and 'Escape' when you are editing colours. 
+
+<a name="AudioGraph"></a>
+
+## Editing the audio graph
+Each and every instrument that you compile in Cabbage will be added to an underlying audio graph. This is true for both Cabbage files, and traditional Csound files. To edit the graph one can launch the Cabbage patcher from the view menu. 
+
+![](../resources/images/10-b-cabbagesynthgraph.gif)
+
+Instruments can also be added directly through the graph by right-clicking and adding them from the context menu. The context menu will show all the examples files, along with a selection of files from a user-defined folder. See the section below on *Settings* to learn how to set this folder. 
+
+![](../resources/images/10-b-cabbageaudiographadd.gif)
+
+Instruments can also be deleted by right-clicking the instrument node. Users can delete/modify connections by clicking on the connections themselves. They can also connect node by clicking and dragging from an output to an input.
+
+![](../resources/images/10-b-cabbageaudiographmodify.gif) 
+
+Once an instrument node has been added, Cabbage will automatically open the corresponding code. Each time you update the corresponding source code, the node will also be updated. 
+
+>As mentioned above, closing a file will not stop it from performing. It is possible to have many instruments running even though their code is not showing. To stop an instrument you must hit the Stop button at the top of its editor, or delete the plugin from the graph.
+
+<a name="Navigating"></a>
+
+## Navigating large source files
+It can become quite tricky to navigate very long text files. For this reason Cabbage provides a means of quickly jumping to instrument definitions. It is also possible to create a special `;- Region:` tag. Any text that appears after this tag will be appended to the drop-down combo box in the Cabbage tool bar.
+![](../../images/docs/navigate_code.gif) 
```

Comment:
This image is missing

Reply:
Thanks, I'll sort that when I get a chance.